src/ops/commit.rs

```diff
@@ -89,3 +92,36 @@ fn commit_fixup_cmd(args: &[OsString], rev: &OsStr) -> Command {
     cmd.args(args);
     cmd
 }
+
+#[derive(Display)]
+#[display(fmt = "Commit instant fixup")]
+pub(crate) struct CommitInstantFixup;
+impl OpTrait for CommitInstantFixup {
+    fn get_action(&self, target: Option<&TargetData>) -> Option<Action> {
+        match target {
+            Some(TargetData::Commit(r)) => {
+                let rev = OsString::from(r);
+
+                Some(Rc::new(move |state: &mut State, term: &mut Term| {
+                    let args = state.pending_menu.as_ref().unwrap().args();
+
+                    state.close_menu();
+
+                    state.run_cmd_interactive(term, commit_fixup_cmd(&args, &rev))?;
+                    state.run_cmd_interactive(
+                        term,
+                        rebase_interactive_cmd(
+                            &args,
+                            &rev,
+                            Some([(OsStr::new("GIT_SEQUENCE_EDITOR"), OsStr::new(":"))].into()),
```

Comment:
Awesome, I will try to get back on it asap.

Reply:
I did copy the functions, and added the tests, and it's working …

But honestly I think ther's a lot of duplication.

Which makes me wonder:

1. Why pass commands instead of libgit2 bindings https://github.com/rust-lang/git2-rs ? It seems  that you're paying needless I/O overhead … you might lose a bit of pretty-printing of commands, I haven't checked the details of the bindings.
2. Why not refactor git-the-command operations and reuse them, because copy-pasta is a guaranteed source of headache

Reply:
Regarding invoking libgit2 vs calling git as a process:
I've tested both methods and I've concluded that in general
- using libgit2 for reading the repo is great, just easier and faster (no need to parse diffs etc. although the library is a bit wonky).
- I recall using libgit2 for "actions" like push/pull or applying patches proved to be cumbersome. And I wanted git to behave as the CLI is already configured by the user.
- Though, it's nice to display exactly what is being executed to the user in a log.

And for re-using commands, I think this is one of few exceptions where a command actually is re-used. But it felt like in this case it wouldn't quite be equal to the other use-case either, as it's not accepting an "args" parameter (user-input params from the menu).

Reply:
> Regarding invoking libgit2 vs calling git as a process: I've tested both methods and I've concluded that in general
> 
>     * using libgit2 for reading the repo is great, just easier and faster (no need to parse diffs etc. although the library is a bit wonky).
>
>     * I recall using libgit2 for "actions" like push/pull or applying patches proved to be cumbersome. And I wanted git to behave as the CLI is already configured by the user.

Very interesting. I wanted to re-write a small tool I made (currently in bash) to compile tree-sitter parsers, and I don't want any modifications, so I guess I'll try this route.

>     * Though, it's nice to display exactly what is being executed to the user in a log.
> 
> 
> And for re-using commands, I think this is one of few exceptions where a command actually is re-used. But it felt like in this case it wouldn't quite be equal to the other use-case either, as it's not accepting an "args" parameter (user-input params from the menu).

Fair enough.